examples/acl.rs

```diff
@@ -19,15 +19,32 @@ fn get_current_user(ctx: &Context, _args: Vec<ValkeyString>) -> ValkeyResult {
     Ok(ValkeyValue::BulkValkeyString(ctx.get_current_user()))
 }
 
+fn custom_category(_ctx: &Context, _args: Vec<ValkeyString>) -> ValkeyResult { 
+    VALKEY_OK
+}
+fn custom_categories(_ctx: &Context, _args: Vec<ValkeyString>) -> ValkeyResult { 
+    VALKEY_OK
+}
+fn existing_categories(_ctx: &Context, _args: Vec<ValkeyString>) -> ValkeyResult { 
+    VALKEY_OK
+}
+
 //////////////////////////////////////////////////////
 
 valkey_module! {
     name: "acl",
     version: 1,
     allocator: (ValkeyAlloc, ValkeyAlloc),
-    data_types: [],
+    data_types: [],    
+    custom_acl_categories: [
+        "custom_acl_one",
+        "custom_acl_two"
+    ]
     commands: [
-        ["verify_key_access_for_user", verify_key_access_for_user, "", 0, 0, 0],
-        ["get_current_user", get_current_user, "", 0, 0, 0],
+        ["verify_key_access_for_user", verify_key_access_for_user, "", "", 0, 0, 0],
+        ["get_current_user", get_current_user, "", "", 0, 0, 0],
+        ["custom_category", custom_category, "write", "custom_acl_one", 0, 0, 0],
+        ["custom_categories", custom_categories, "", "custom_acl_one custom_acl_two", 0, 0, 0],
+        ["existing_categories", existing_categories, "write", "read fast admin", 0, 0, 0],
```

Comment:
```suggestion
        ["existing_categories", existing_categories, "write", "read fast admin", 0, 0, 0],
```
Is there any way to make "new categories" be optional so that folks don't have to change existing code to adopt this?

Reply:
Yes, I was looking into this with Cameron, we can use the `$(, $optional_arg:expr)?` style macro field updates to make this a completely optional change.

It is possible to make it such it existing code changes does not need to adopt it; We tried it out.

Cameron will push out a commit soon

Reply:
Commit is now out with it being an optional change so people don't have to actually update anything